esp-rtos/src/run_queue.rs

```diff
@@ -60,7 +70,9 @@ impl RunQueue {
         rtos_trace::trace::task_ready_begin(ready_task.rtos_trace_id());
 
         self.ready_priority.mark_ready(priority);
-        if priority > current_prio || current_prio == 0 {
+
+        let current_minimum_prio = self.current_prios.iter().copied().min().unwrap();
+        if priority > current_minimum_prio || current_minimum_prio == 0 {
```

Comment:
At this point we should almost complicate this a bit further - we should take core affinity into account, and we can tell which core `ready_task` can be scheduled on. This could replace much of the `Scheduler::trigger_schedule` logic, which function is always called depending on the result of `mark_task_ready` anyway.